File: src/desktop/Level/Clue/weapon_caution_clue.py
from subclue import SubClue

class WeaponCautionClue(SubClue):
    """ Represents the Weapon Caution Clue """
    
    def __init__(self):
        """ Initialize the Clue """
        self.nearWeapons = False
        self.reverse = False
    
    def updateFromAdjacentSquare(self, square):
        """ Update the adjacency clue with an adjacent square """
        if square.hasWeaponry():
            self.nearWeapons = True 

File: src/android/src/com/kaottessur/pymine/level/clue/WeaponCautionClue.java
package com.kaottessur.pymine.level.clue;

import com.kaottessur.pymine.level.GridSquare;
import com.kaottessur.pymine.level.Minefield;

public class WeaponCautionClue extends SubClue {
	private boolean nearWeapons = false;
	
	public boolean isNearWeaponry() {
		return nearWeapons;
	}
	
	@Override
	public void update(Minefield minefield, int gridRow, int gridColumn) {
		nearWeapons = false;
	}
	
	@Override
	public void updateFromAdjacentSquare(GridSquare gridSquare) {
		if (gridSquare.mined()) {
			nearWeapons = true;
		}
	}
}
